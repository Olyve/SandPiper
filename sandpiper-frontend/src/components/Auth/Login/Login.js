import React, { Component } from 'react';
import './Login.css';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };

    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  onChange(event) {
    let target = event.target;
    let value = target.value;
    let name = target.name;

    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    this.props.handleLogin({
      email: this.state.email,
      password: this.state.password
    });
    this.setState({
      email: '',
      password: ''
    });

    event.preventDefault();
  }

  render() {
    return (
      <div className='login'>
        <h2>Login</h2>
        <form>
          <div className='inputs'>
            <div>
              <label className='inputs-label'><span>Email: </span>
                <input type='text' name='email' value={this.state.email} onChange={this.onChange} />
              </label>
            </div>
            <div>
              <label className='inputs-label'><span>Password: </span>
                <input type='password' name='password' value={this.state.password} onChange={this.onChange} />
              </label>
            </div>
            <div className='inputs-submit'>
              <button className='submit' onClick={this.handleSubmit}>Submit</button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export default Login;