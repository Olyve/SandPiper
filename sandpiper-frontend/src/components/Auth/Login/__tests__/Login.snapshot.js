import React from 'react';
import Login from '../Login';

describe('Login', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(
      <Login />
    );

    expect(wrapper).toMatchSnapshot();
  });
});