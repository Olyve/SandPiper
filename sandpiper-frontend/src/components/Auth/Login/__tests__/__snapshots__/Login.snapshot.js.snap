// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render without crashing 1`] = `
<div
  className="login"
>
  <h2>
    Login
  </h2>
  <form>
    <div
      className="inputs"
    >
      <div>
        <label
          className="inputs-label"
        >
          <span>
            Email: 
          </span>
          <input
            name="email"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
      </div>
      <div>
        <label
          className="inputs-label"
        >
          <span>
            Password: 
          </span>
          <input
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </label>
      </div>
      <div
        className="inputs-submit"
      >
        <button
          className="submit"
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
